src/main/java/com/spotify/reaper/resources/ClusterResource.java

```diff
@@ -294,26 +305,64 @@ public Response deleteCluster(
-      throws ReaperException {
-    Optional<String> allEndpointsState = Optional.absent();
+  /**
+   * Callable to get and parse endpoint states through JMX
+   *
+   *
+   * @param seedHost The host address to connect to via JMX
+   * @return An optional NodesStatus object with the status of each node in the cluster as seen from the seedHost node
+   */
+  Callable<Optional<NodesStatus>> getEndpointState(String seedHost) {
+    return () -> {
+      try (JmxProxy jmxProxy = context.jmxConnectionFactory.connect(seedHost)) {
+        Optional<String> allEndpointsState = Optional.fromNullable(jmxProxy.getAllEndpointsState());
+        return Optional.of(new NodesStatus(seedHost, allEndpointsState.or("")));
+      } catch (Exception e) {
+        LOG.debug("failed to create cluster with seed host: {}", seedHost, e);
+        Thread.sleep(TimeUnit.MILLISECONDS.convert(JmxProxy.JMX_CONNECTION_TIMEOUT, JmxProxy.JMX_CONNECTION_TIMEOUT_UNIT));
+        return Optional.absent();
+      }
+    };
+  }
+  
+  
+  /**
+   * Get all nodes state by querying the AllEndpointsState attribute through JMX.
+   * 
+   * To speed up execution, the method calls JMX on 3 nodes asynchronously and processes the first response
+   * 
+   * @param cluster
+   * @return An optional NodesStatus object with all nodes statuses
+   */
+  public Optional<NodesStatus> getNodesStatus(Optional<Cluster> cluster){
     Optional<NodesStatus> nodesStatus = Optional.absent();
     if(cluster.isPresent() && cluster.get().getSeedHosts()!=null) {
-      for(String seedHost:cluster.get().getSeedHosts()) {
-        try (JmxProxy jmxProxy = context.jmxConnectionFactory.connect(seedHost)) {
-          allEndpointsState = Optional.fromNullable(jmxProxy.getAllEndpointsState());
-          if (allEndpointsState.isPresent()) {
-            nodesStatus = Optional.of(new NodesStatus(seedHost, allEndpointsState.or("")));
-            break;
-          }
-        } catch (ReaperException e) {
-          LOG.error("failed to create cluster with seed host: {}", seedHost, e);
+      
+      List<String>seedHosts = Lists.newArrayList();
+      seedHosts.addAll(cluster.get().getSeedHosts());
+      Collections.shuffle(seedHosts);
+      List<List<String>> partitionedSeedList = Lists.partition(seedHosts, JMX_NODE_STATUS_CONCURRENCY);
+      
+      for(List<String> seeds:partitionedSeedList) {
+        LOG.info("seed list : {}", seeds);
+        List<Callable<Optional<NodesStatus>>> endpointStateTasks = seeds.stream()
+                                                                   .map(seedHost -> getEndpointState(seedHost))
+                                                                   .collect(Collectors.toList());
+        
+        
+        try {
+          nodesStatus = clusterStatusExecutor.invokeAny(endpointStateTasks, JmxProxy.JMX_CONNECTION_TIMEOUT, JmxProxy.JMX_CONNECTION_TIMEOUT_UNIT);
+        } catch (Exception e) {
+          // TODO Auto-generated catch block
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)